### PR TITLE
Feat code improvements

### PR DESCRIPTION
**Pull request by claudiaelizabete**
#### **Overview**

- This pull request introduces a link attacher package forked from https://github.com/davidorchansky/gazebo_ros_link_attacher. Some modifications were made to the code style to bring it within the standards.

#### **Use the following settings in the overrides file for docker testing**

- Use these settings to compile and run the module using `autoproj`.
- The contents of the files manifest and `overrides.yaml` must be changed according to the `templates` below.
- Additional packages may be needed to test this module.

  e.g.,

- _overrides.yaml_

  ```yaml
    overrides:
      - gazebo_ros_link_attacher:
        branch: feat-code-improvements
      - pkg_set:github:/Brazilian-Institute-of-Robotics/cbr_package_set:
        branch: feat-link-attacher
  ```

- _manifest_

  ```yaml
      package_sets:
        - github: Brazilian-Institute-of-Robotics/ros2.gazebo-package_set
          branch: main
          private: true
        - github: Brazilian-Institute-of-Robotics/ros2.distro-package_set
          branch: main
          private: true
        - github: Brazilian-Institute-of-Robotics/core_base-package_set
          branch: main
          private: true
        - github: Brazilian-Institute-of-Robotics/cbr_package_set
          branch: main
          private: true

      layout:
        - gazebo_ros_link_attacher
  ```

#### **What was added/changed in this update**

- [Code style improvements](https://github.com/Brazilian-Institute-of-Robotics/gazebo_ros_link_attacher/commit/e2b00eb1ef5ecf9ab49e548ba4d2b1ada2755c9e)

#### **Depends On:**

- N/A

#### **Related Issues:**

- N/A

#### **Notes:**

- N/A

### CODE REVIEW

**Review by Lecosant — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -8,53 +8,59 @@ cd gazebo_link_attacher_ws/src
 catkin_init_workspace
 git clone https://github.com/pal-robotics/gazebo_ros_link_attacher.git
```
Comment (rodrigoformiga) [inline README.md]: git clone git@github.com:Brazilian-Institute-of-Robotics/gazebo_ros_link_attacher.git

File: README.md [hunk 2]
```diff
@@ -8,53 +8,59 @@ cd gazebo_link_attacher_ws/src
...
 cd ..
-catkin_make
-source devel/setup.bash
+
+rosdep update
+rosdep install --from-paths src --ignore-src -y --rosdistro humble
```
Comment (rodrigoformiga) [inline README.md]: ERROR: gazebo_ros_link_attacher: Cannot locate rosdep definition for [transforms3d]

**Review by rodrigoformiga — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (claudiaelizabete) [reply README.md]: done in https://github.com/Brazilian-Institute-of-Robotics/gazebo_ros_link_attacher/pull/1/commits/256f02a87d17adddfcafe394aa3c7e56aa83a5bf

File: README.md (hunk 2 above)
Comment (claudiaelizabete) [reply README.md]: done in https://github.com/Brazilian-Institute-of-Robotics/cbr_package_set/commit/b1f78fc1528cf446747181f2f95fce637f47f7a5

File: include/gazebo_ros_link_attacher/gazebo_ros_link_attacher.h [hunk 3]
```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2024, SENAI Cimatec
```
Comment (qleonardolp) [inline include/gazebo_ros_link_attacher/gazebo_ros_link_attacher.h]: The `uncrustify` check conflicts with `cpplint` about this file. It's not a critical error, but it is the root of the error on the tests with colcon. Tests with autoproj are passing completely, though. One can check it by:
Formatting with uncrustify:
```bash
ament_uncrustify --reformat include/gazebo_ros_link_attacher/gazebo_ros_link_attacher.h
```
Check with cpplint:
```bash
ament_cpplint
```

File: scripts/attach.py
Comment (qleonardolp) [inline scripts/attach.py:1]: Format python files with `ament_black`

File: scripts/demo.py
Comment (qleonardolp) [inline scripts/demo.py:1]: I recommend to change this folder from "scripts" to "examples"

File: src/gazebo_ros_link_attacher.cpp [hunk 4]
```diff
@@ -1,211 +1,256 @@
...
+  fixed_joint.joint = this->physics->CreateJoint("revolute", m1);
+  this->joints_.push_back(fixed_joint);
+
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "Attach");
+  fixed_joint.joint->Attach(l1, l2);
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "Loading links");
+  fixed_joint.joint->Load(l1, l2, ignition::math::Pose3d());
+  RCLCPP_DEBUG_STREAM(node_->get_logger(), "SetModel");
+  fixed_joint.joint->SetModel(m2);
+  /*
```
Comment (qleonardolp) [inline src/gazebo_ros_link_attacher.cpp:147]: I recommend removing this comment and using the readme section _Bugs & Feature Requests_.  Include a subsection there such as 'Konwn Issue" to report about the SetModel method.

**Review by qleonardolp — CHANGES_REQUESTED** [review]

File: include/gazebo_ros_link_attacher/gazebo_ros_link_attacher.h (hunk 3 above)
Comment (claudiaelizabete) [reply include/gazebo_ros_link_attacher/gazebo_ros_link_attacher.h]: done in https://github.com/Brazilian-Institute-of-Robotics/gazebo_ros_link_attacher/commit/d1afd8f98b012a6dadaa6aefd431bcd1c6dbcefd

File: scripts/demo.py
Comment (claudiaelizabete) [reply scripts/demo.py:1]: done in https://github.com/Brazilian-Institute-of-Robotics/gazebo_ros_link_attacher/commit/d1afd8f98b012a6dadaa6aefd431bcd1c6dbcefd

File: scripts/attach.py
Comment (claudiaelizabete) [reply scripts/attach.py:1]: for now, I'll maintain the formatting using only flake8 and pep257.

**Review by qleonardolp — APPROVED** [review]

**Pull request merged**